src/cowboy_req.erl

```diff
@@ -1222,26 +1222,43 @@ response_merge_headers(Headers, RespHeaders, DefaultHeaders) ->
 		merge_headers(Headers2, RespHeaders),
 		DefaultHeaders).
 
+camelize_header({N,V}) ->
+    case N of
+        <<"ETag">> ->
+            {N,V};
+        <<"x-amz-meta", _/binary>> ->
+            {N,V};
+        N when is_binary(N) ->
+            {cowboy_bstr:capitalize_token(N), V};
+        _ ->
+            {N,V}
+    end.
+
 -spec merge_headers(cowboy:http_headers(), cowboy:http_headers())
 	-> cowboy:http_headers().
 
+merge_headers(Headers_1, Headers_2) ->
+    CHeaders_1 = [camelize_header(E) || E <- Headers_1],
+    CHeaders_2 = [camelize_header(E) || E <- Headers_2],
+    merge_headers_1(CHeaders_1, CHeaders_2).
+
 %% Merge headers by prepending the tuples in the second list to the
 %% first list. It also handles Set-Cookie properly, which supports
 %% duplicated entries. Notice that, while the RFC2109 does allow more
 %% than one cookie to be set per Set-Cookie header, we are following
 %% the implementation of common web servers and applications which
 %% return many distinct headers per each Set-Cookie entry to avoid
 %% issues with clients/browser which may not support it.
-merge_headers(Headers, []) ->
+merge_headers_1(Headers, []) ->
```

Comment:
Drop a comment why converting Camel Case is needed.
Adding a link to the issue on github would be enough to me.
https://github.com/leo-project/leofs/issues/489